plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2030,12 +2030,29 @@ int getReservedCpuMHZ(VirtualMachineTO vmSpec) {
         return new String[] {datastoreDiskPath};
     }
 
-    // Pair<internal CS name, vCenter display name>
-    private Pair<String, String> composeVmNames(VirtualMachineTO vmSpec) {
-        String vmInternalCSName = vmSpec.getName();
-        String vmNameOnVcenter = vmSpec.getName();
-        if (_instanceNameFlag && vmSpec.getHostName() != null) {
-            vmNameOnVcenter = vmSpec.getHostName();
+
+    /**
+     * This method generates VM name for Vcenter and Cloudstack( when Hypervisor is VMware).
+     * It generates VM name according to  _instanceNameFlag setting.
+     *
+     * @param VirtualMachineTO
+     *        vmSpec
+     * @return Pair<internal CS name, vCenter display name>. A pair which contain 'internal CS name'  and
+     *         'vCenter display name'(vCenter knows VM by this name).
+     **/
+    private Pair<String, String> composeVmNatmes(VirtualMachineTO vmSpec) {
+
+        String vmInternalCSName = null;
```

Comment:
Try to use emtpy/blank rather than null here, because these are being returned. Good practice to avoid returning nulls.
Of course, empty vm name is obviously invalid both in ACS database as well as hypervisor (VMware here, this code is not being invoked for other hypervisor any way)